pkg/apis/flowcontrol/v1beta1/defaults.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"k8s.io/api/flowcontrol/v1beta1"
+)
+
+// Default settings for flow-schema
+const (
+	FlowSchemaDefaultMatchingPrecedence int32 = 1000
+)
+
+// Default settings for priority-level-configuration
+const (
+	PriorityLevelConfigurationDefaultHandSize                 int32 = 8
+	PriorityLevelConfigurationDefaultQueues                   int32 = 64
+	PriorityLevelConfigurationDefaultQueueLengthLimit         int32 = 50
+	PriorityLevelConfigurationDefaultAssuredConcurrencyShares int32 = 30
+)
+
+// SetDefaults_FlowSchema sets default values for flow schema
+func SetDefaults_FlowSchemaSpec(spec *v1beta1.FlowSchemaSpec) {
+	if spec.MatchingPrecedence == 0 {
+		spec.MatchingPrecedence = FlowSchemaDefaultMatchingPrecedence
+	}
+	// For data-plane 1.18, 1.19 compatibility
```

Comment:
can you clarify what you mean by this? SetDefaults_ResourcePolicyRule does not do anything

Reply:
That function is a vestige of a more ambitious PR (which adds another condition to a ResourceRule).  It could easily be trimmed from this PR.